interfaces/datastructure/Mask2D.h

```diff
@@ -0,0 +1,165 @@
+/**
+ * @copyright Copyright (c) 2017-2025 B-com http://www.b-com.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MASK2D_H
+#define MASK2D_H
+
+#include "core/SerializationDefinitions.h"
+#include "core/SolARFrameworkDefinitions.h"
+#include "datastructure/Image.h"
+#include <map>
+#include <string>
+
+namespace SolAR {
+namespace datastructure {
+
+/**
+ * @enum class Segmentation2DType
+ */
+enum class Segmentation2DType {
+    INSTANCE,
+    PANOPTIC,
+    SEMANTIC,
+    UNDEFINED
+};
+static const std::map<Segmentation2DType, std::string> segmentation2DTypeToStr {
```

Comment:
`static std::string toString(const Segmentation2DType segmentation2DType)` to be complient with other classes?